### ISSUE TITLE
Null pointer exception in dynamic pure loop cutting

### ISSUE THREAD

**Issue by hernanponcedeleon** [hernanponcedeleon/Dat3M#583]
```
./Dartagnan-SVCOMP.sh ../../sv-benchmarks/c/properties/no-data-race.prp ../../sv-benchmarks/c/pthread-atomic/lamport-b.i


--------------------------------------------------------------------------------


17:19:49 [INFO ] GitInfo.CreateGitInfo - Git branch: svcomp-2024
17:19:49 [INFO ] GitInfo.CreateGitInfo - Git commit ID: 17eeec33cbc9c310519b08551b317469f878ce8c
17:19:49 [INFO ] GlobalSettings.LogGlobalSettings - REFINEMENT_GENERATE_GRAPHVIZ_DEBUG_FILES: false
17:19:49 [INFO ] GlobalSettings.LogGlobalSettings - REFINEMENT_SYMMETRIC_LEARNING: FULL
17:19:49 [INFO ] Dartagnan.main - Program path: /tmp/vcloud_worker_vcloud-master_on_vcloud-master/run_dir_efa08a41-e1d0-40b1-986e-392aa48b4f7c/bin/dartagnan-verify-smqyFgON4Q/output/lamport-b-opt.ll
17:19:49 [INFO ] Dartagnan.main - CAT file path: cat/svcomp.cat
17:19:50 [INFO ] Wmm.configureAll - encoding.wmm.reduceAcyclicityEncodeSets: true
17:19:50 [INFO ] LoopFormVerification.run - Detected 8 loops in the program.
17:19:50 [INFO ] Compilation.run - Program compiled to C11
17:19:50 [INFO ] LoopUnrolling.run - Program unrolled 1 times
17:19:50 [ERROR] Dartagnan.main - Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.Map.get(Object)" is null
java.lang.NullPointerException: Cannot invoke "java.lang.Integer.intValue()" because the return value of "java.util.Map.get(Object)" is null
	at com.dat3m.dartagnan.program.processing.DynamicPureLoopCutting.lambda$computeDominatorTree$8(DynamicPureLoopCutting.java:332) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.program.processing.DynamicPureLoopCutting.commonDominator(DynamicPureLoopCutting.java:357) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.program.processing.DynamicPureLoopCutting.lambda$computeDominatorTree$9(DynamicPureLoopCutting.java:341) ~[dartagnan.jar:?]
	at java.util.stream.ReduceOps$1ReducingSink.accept(ReduceOps.java:80) ~[?:?]
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?]
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
	at java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
	at java.util.stream.ReferencePipeline.reduce(ReferencePipeline.java:657) ~[?:?]
	at com.dat3m.dartagnan.program.processing.DynamicPureLoopCutting.computeDominatorTree(DynamicPureLoopCutting.java:341) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.program.processing.DynamicPureLoopCutting.reduceToDominatingSideEffects(DynamicPureLoopCutting.java:275) ~[dartagnan.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.dat3m.dartagnan.program.processing.DynamicPureLoopCutting.run(DynamicPureLoopCutting.java:77) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.program.processing.ProcessingManager.lambda$run$0(ProcessingManager.java:142) ~[dartagnan.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at com.dat3m.dartagnan.program.processing.ProcessingManager.run(ProcessingManager.java:142) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.verification.solving.ModelChecker.preprocessProgram(ModelChecker.java:71) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.verification.solving.DataRaceSolver.run(DataRaceSolver.java:50) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.verification.solving.DataRaceSolver.run(DataRaceSolver.java:40) ~[dartagnan.jar:?]
	at com.dat3m.dartagnan.Dartagnan.main(Dartagnan.java:157) [dartagnan.jar:?]
ERROR
```

**Comment by ThomasHaas** [comment]
This was fixed by #619, right?

**Comment by hernanponcedeleon** [comment]
Yes
